src/Parser/AfflictionWarlock/Modules/Items/Legendaries/TheMasterHarvester.js

```diff
@@ -0,0 +1,51 @@
+import Module from 'Parser/Core/Module';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import { formatNumber } from 'common/format';
+
+import getDamageBonus from '../../WarlockCore/getDamageBonus';
+
+const SOUL_HARVEST_DAMAGE_BONUS = .2;
+
+class TheMasterHarvester extends Module {
+  playerBonusDmg = 0;
+  petBonusDmg = 0;
+
+  petIds = [];
+  on_initialized() {
+    if (!this.owner.error) {
+      this.active = this.owner.selectedCombatant.hasChest(ITEMS.THE_MASTER_HARVESTER.id);
+    }
+    this.owner.report.friendlyPets.filter(pet => pet.petOwner === this.owner.playerId).forEach(pet => {
```

Comment:
That's probably right, I don't yet have a log that I could test this on, but my guess is that all pets summoned throughout the instance (since that's what is in this.report, no?) should be in the in the array, and re-summoned pets should have the same id. I would test this on training dummy myself but WCL treats it as a trash fight and so Analyzer doesn't even show it. Some LFR maybe. I'll look into it (same problem will be happening with Soul Harvest as it buffs pet's damage too)